K2Bridge/KustoDAL/KustoDataAccess.cs

```diff
@@ -147,22 +152,96 @@ public async Task<ResolveIndexResponse> ResolveIndexAsync(string indexName)
+                $@"{KustoQLOperators.Let} percentage = {DynamicSamplePercentage} / 100.0;
+{KustoQLOperators.Let} table_count = {KustoQLOperators.ToScalar}({tableName} | {KustoQLOperators.Count});
+{tableName} | {KustoQLOperators.Sample} {KustoQLOperators.ToInt}({KustoQLOperators.Floor}(table_count * percentage, 1)) | {KustoQLOperators.Summarize} {KustoQLOperators.BuildSchema}({fieldCapabilityElement.Name})" :
+                $"{tableName} | {KustoQLOperators.Summarize} {KustoQLOperators.BuildSchema}({fieldCapabilityElement.Name})";
+            var (_, result) = await Kusto.ExecuteQueryAsync(new QueryData(query, tableName), RequestContext);
+            result.Read();
+            var jsonResult = result[0];
+            var stack = new Stack<(string, JToken)>();
+            stack.Push((fieldCapabilityElement.Name, (JToken)jsonResult));
 
-                Logger.LogDebug("Found field: {@fieldCapabilityElement}", fieldCapabilityElement);
+            while (stack.Count > 0)
+            {
+                var (name, jsonObject) = stack.Pop();
+                /* The special name `indexer` indicates that the field is an array.
+                    * In kibana, there is no difference between an array field and a normal field (for example, any int field can contain a single int value or an array of int values).
+                    */
+                const string specialArrayKey = "`indexer`";
+                switch (jsonObject)
+                {
+                    case JValue v:
+                        AddSingleDynamicField(response, name, v);
+                        break;
+                    case JArray arr:
+                        AddSingleDynamicField(response, name, arr);
+                        break;
+                    case JObject obj when obj[specialArrayKey] != null:
+                        AddSingleDynamicField(response, name, obj[specialArrayKey]);
+                        break;
+                    case JObject obj:
+                        AddSingleDynamicField(response, name, "dynamic");
+                        foreach (var property in obj.Properties())
+                        {
+                            stack.Push((name + "." + property.Name, property.Value));
+                        }
+
+                        break;
+
+                    default:
+                        throw new InvalidOperationException($"Unexpected type {jsonResult.GetType()}");
+                }
             }
         }
 
+        private void AddSingleDynamicField(FieldCapabilityResponse response, string name, JToken type)
+        {
+            var newField = FieldCapabilityElementFactory.CreateFromNameAndKustoShorthandType(name, CombineValues(type));
+            Logger.LogDebug("Added dynamic field @{newField} ", newField);
+            response.AddField(newField);
+        }
+
+        private string CombineValues(JToken property)
+        {
+            return property is JArray ? "string" : property.ToString();
```

Comment:
I have a small sample dataset with a dynamic column and a case that might be worth considering: two "compatible" numeric types could be converted into e.g. `double`. Otherwise, these fields end up as strings and we can't do any metrics on them...

Sample schema:

```json
{{
  "usage_idle": [
    "double",
    "long"
  ],
  "usage_guest": "long",
  "usage_guest_nice": "long",
  "usage_nice": "long",
  "usage_system": [
    "double",
    "long"
  ],
  "usage_iowait": "long",
  "usage_softirq": "long",
  "usage_irq": "long",
  "usage_steal": "long",
  "usage_user": [
    "double",
    "long"
  ]
}}
```

Reply:
Let's revisit in a future user story.